### PR TITLE
Fix issue with pyserial3.5 and remove pinned requirement pyserial<3.5

### PR DESCRIPTION
**Pull request by trappitsch**
Adjusted the mocker that faked serial ports for appropriate initialization, see code comment below.

This should take care of #298

### CODE REVIEW

File: instruments/tests/test_base_instrument.py [hunk 1]
```diff
@@ -141,17 +141,15 @@ def fake_comports():
     """
     Generate a fake list of comports to compare against.
     """
-    fake_device = ListPortInfo()
+    fake_device = ListPortInfo(device='COM1')
     fake_device.vid = 0
     fake_device.pid = 1000
     fake_device.serial_number = 'a1'
-    fake_device.device = 'COM1'
```
Comment (trappitsch) [inline instruments/tests/test_base_instrument.py:148]: Device must now be initialized with the class. See explanation in #298. This is also compatible with pyserial3.4.

3 more identical changes throughout mocked classes.

File: setup.cfg [hunk 2]
```diff
@@ -28,7 +28,7 @@ classifiers =
 include_package_data = True
 packages = find:
 install_requires =
-    pyserial>=3.3,<3.5
+    pyserial>=3.3
```
Comment (trappitsch) [inline setup.cfg:31]: Change version pinning back to previous value.

**Comment by trappitsch** [comment]
> Coverage decreased (-0.0003%) to 98.85%

Not really sure why the coverage seems to have decreased... The file has two lines less, so mathematically it should go up, no?

**Comment by scasagrande** [comment]
Thanks for fixing this @trappitsch :D

**Pull request merged**